### PR TITLE
Make gamelab animation list resilient to multiple keys.

### PR DESCRIPTION
**Pull request by epeach**
In response to Zendesk tickets coming in with reports of student projects not loading due to duplicated keys in the animation list. This PR fixes the symptom, but not the cause. Cause is currently under-investigation.

Before: Throw error.
Now: Don't throw error - instead repair animation list by removing duplicates.

Includes tests.

### CODE REVIEW

File: apps/test/unit/gamelab/shapesTest.js [hunk 1]
```diff
@@ -0,0 +1,58 @@
+import {throwIfSerializedAnimationListIsInvalid} from '@cdo/apps/gamelab/shapes';
```
Comment (caleybrock) [inline apps/test/unit/gamelab/shapesTest.js:1]: is this still a good name for this function?

Comment (epeach) [reply apps/test/unit/gamelab/shapesTest.js:1]: That's a good question! After some thought, I think it still makes sense. The majority of this function is throwing errors if the list is invalid. I also think it makes sense to keep this de-duplication in the function as it is part of the validation of the list. I think the purpose of the function remains unchanged - before and after this PR, if this function is called with a list and doesn't throw an error, then we know that the list is valid.

Comment (caleybrock) [reply apps/test/unit/gamelab/shapesTest.js:1]: great, thanks for being so thoughful

File: apps/src/gamelab/shapes.js [hunk 2]
```diff
@@ -182,13 +182,20 @@ export function throwIfSerializedAnimationListIsInvalid(serializedAnimationList)
...
+      // Store indices from highest to lowest to make it easier to remove dupes
+      dupIndices.splice(0, 0, index);
     }
     knownKeys[key] = true;
   });
 
+  // Remove any duplicated indices
+  dupIndices.forEach(index => {
+    orderedKeys.splice(index, 1);
+  });
```
Comment (davidsbailey) [inline apps/src/gamelab/shapes.js]: your code here looks good, but for utility method operations like this, I'd always rather use an existing library, in this case `_.uniq`: https://lodash.com/docs/4.17.5#uniq

**Pull request merged**